tools/flaky_tests/Jenkinsfile

```diff
@@ -0,0 +1,74 @@
+// -*- mode: groovy -*-
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+err = null
+
+node('mxnetlinux-cpu') {
+  // Loading the utilities requires a node context unfortunately
+  checkout scm
+  utils = load('ci/Jenkinsfile_utils.groovy')
+}
+utils.assign_node_labels(linux_gpu: 'mxnetlinux-gpu')
+tests = false
+
+utils.main_wrapper(
+core_logic: {
+    stage('Preprocessing'){
+        node(NODE_LINUX_CPU){
+            ws('workspace/fc-preprocessing'){
+                utils.init_git()
+                utils.docker_run('ubuntu_cpu', 'select_tests', false)
+                tests = fileExists('tests.tmp')
+                stash name:'flaky_tests', includes:'tests.tmp' 
+            }
+        }
+    }
+
+    // only continue if some tests were selected
+    if( ! tests ) {
+        currentBuild.result = 'SUCCESS'
+        return
```

Comment:
Please don't return here. The wrapping logic (utils.main_wrapper) is taking care of propagating the results properly. Just do nothing in that case.